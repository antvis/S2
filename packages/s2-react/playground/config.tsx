/* eslint-disable no-console */
import {
  ResizeType,
  customMerge,
  type CustomHeaderField,
  type CustomTreeNode,
  type S2DataConfig,
  type S2TableSheetFrozenOptions,
  type ThemeCfg,
} from '@antv/s2';
import { getBaseSheetComponentOptions } from '@antv/s2-shared';
import { PivotSheetMultiLineTextDataCfg } from '@antv/s2/__tests__/data/data-multi-line-text';
import type { SliderSingleProps } from 'antd';
import {
  data,
  fields,
  meta,
  totalData,
} from '../__tests__/data/mock-dataset.json';
import type { SheetComponentOptions } from '../src/components';

export const tableSheetSingleColumns: CustomHeaderField[] = [
  'province',
  'city',
  'type',
  'sub_type',
  'number',
];

export const tableSheetMultipleColumns: CustomTreeNode[] = [
  {
    field: 'area',
    title: '地区地区地区地区地区地区地区地区地区',
    children: [
      {
        field: 'province',
        title: '省份省份省份省份省份省份省份省份省份',
        children: [
          {
            field: 'type',
            title: '类型类型类型类型类型类型类型类型类型类型',
          },
        ],
      },
      { field: 'city', title: '城市' },
    ],
  },
  {
    field: 'money',
    title: '金额',
    children: [
      { field: 'price', title: '价格', description: '价格描述' },
      { field: 'number', title: '数量' },
    ],
  },
];

export const tableSheetDataCfg: S2DataConfig = {
  data,
  totalData,
  meta: [
    { field: 'number', name: '数值', formatter: (v) => `${v}-@` },
    ...meta,
  ],
  fields: {
    columns: tableSheetSingleColumns,
  },
};

export const pivotSheetDataCfg: S2DataConfig = {
  data,
  totalData,
  meta,
  fields,
};

export const pivotSheetMultiLineTextDataCfg = PivotSheetMultiLineTextDataCfg;

export const pivotSheetDataCfgForCompactMode = customMerge<S2DataConfig>(
  pivotSheetDataCfg,
  {
    data: [
      ...pivotSheetDataCfg.data,
      {
        province: '浙江',
        city: '杭州',
        type: '笔',
        price: '11111111',
      },
      {
        province: '浙江',
        city: '杭州',
        type: '纸张',
        price: '2',
      },
      {
        province: '浙江',
        city: '舟山',
        type: '笔',
        price: '2',
      },
      {
        province: '浙江',
        city: '舟山',
        type: '纸张',
        price: '133.333',
      },
    ],
  },
);

export const s2ConditionsOptions: SheetComponentOptions['conditions'] = {
  text: [
    {
      field: 'province',
      mapping(value) {
        if (value === '浙江省') {
          return {
            fontSize: 30,
            fill: 'pink',
            textAlign: 'center',
            fontWeight: 'bold',
          };
        }
      },
    },
    {
      field: 'city',
      mapping(value) {
        if (value === '成都市') {
          return {
            fontSize: 20,
            fill: 'red',
            textAlign: 'right',
            fontWeight: 'bold',
          };
        }
      },
    },
    {
      field: 'type',
      mapping() {
        return {
          fontSize: 20,
          fill: 'red',
          textAlign: 'right',
        };
      },
    },
    {
      field: 'sub_type',
      mapping() {
        return {
          fontSize: 12,
          opacity: 0.6,
          fill: 'yellow',
          textAlign: 'left',
        };
      },
    },
    {
      field: 'number',
      mapping(value) {
        if (+value <= 3000) {
          return {
            fill: '#065',
            fontWeight: 800,
            fontSize: 20,
          };
        }

        if (+value > 3000) {
          return {
            fill: '#000',
            opacity: 0.4,
          };
        }

        return {
          fontSize: 30,
          fill: '#000',
          textAlign: 'left',
        };
      },
    },
  ],
  icon: [
    {
      field: 'type',
      position: 'left',
      mapping() {
        return {
          icon: 'CellUp',
          fill: '#054',
        };
      },
    },
    {
      field: 'city',
      position: 'left',
      mapping(value) {
        if (value === '成都市') {
          return {
            icon: 'CellUp',
            fill: '#FF4D4F',
          };
        }
      },
    },
    {
      field: 'province',
      position: 'left',
      mapping() {
        return {
          icon: 'CellUp',
          fill: '#FF4D4F',
        };
      },
    },
    {
      field: 'number',
      mapping() {
        return {
          icon: 'CellUp',
          fill: '#FF4D4F',
        };
      },
    },
  ],
  interval: [
    {
      field: 'number',
      mapping() {
        return {
          fill: '#80BFFF',
          // 自定义柱状图范围
          isCompare: true,
          maxValue: 8000,
          minValue: 300,
        };
      },
    },
  ],
};

export const TableSheetFrozenOptions: S2TableSheetFrozenOptions = {
  colCount: 1,
  trailingColCount: 1,
};

export const S2TooltipOptions: SheetComponentOptions['tooltip'] = {
  operation: {
    menu: {
      // 支持透传 Ant Design <Menu/> 组件 API: https://ant-design.antgroup.com/components/menu-cn#api
      // mode: 'vertical',
      onClick(info, cell) {
        console.log('菜单项点击: ', info, cell);
      },
      items: [
        {
          key: 'custom-a',
          label: '操作1',
          icon: 'Trend',
          onClick: (info, cell) => {
            console.log('操作1点击:', info, cell);
          },
          children: [
            {
              key: 'custom-a-a',
              label: '操作 1-1',
              icon: 'Trend',
              onClick: (info, cell) => {
                console.log('操作1-1点击:', info, cell);
              },
            },
          ],
        },
        {
          key: 'custom-b',
          label: '操作2',
          icon: 'EyeOutlined',
          onClick: (info, cell) => {
            console.log('操作2点击:', info, cell);
          },
        },
        {
          key: 'custom-c',
          label: '操作3',
          icon: 'EyeOutlined',
          visible: false,
          onClick: (info, cell) => {
            console.log('操作3点击:', info, cell);
          },
        },
        {
          key: 'custom-c',
          label: '操作4',
          icon: 'EyeOutlined',
          visible: (cell) => {
            // 叶子节点才显示
            const meta = cell.getMeta();

            return meta.isLeaf;
          },
          onClick: (info, cell) => {
            console.log('操作4点击:', info, cell);
          },
        },
      ],
    },
  },
};

export const s2Options: SheetComponentOptions = {
  debug: true,
  width: 800,
  height: 600,
  hierarchyType: 'grid',
  seriesNumber: {
    enable: true,
  },
  transformCanvasConfig() {
    return {
      supportsCSSTransform: true,
      // devicePixelRatio: 3,
      // cursor: 'crosshair',
    };
  },
  frozen: {
    rowHeader: true,
    // rowCount: 1,
    // trailingRowCount: 1,
    // colCount: 1,
    // trailingColCount: 1,
  },
  cornerText: '测试测试测试测试测试测试测试测试测试测试',
  interaction: {
    copy: {
      enable: true,
      withFormat: true,
      withHeader: true,
    },
    hoverAfterScroll: true,
    hoverHighlight: true,
    selectedCellHighlight: true,
    selectedCellMove: true,
    rangeSelection: true,
    // 防止 mac 触控板横向滚动触发浏览器返回, 和移动端下拉刷新
    overscrollBehavior: 'none',
    brushSelection: {
      dataCell: true,
      colCell: true,
      rowCell: true,
    },
    resize: {
      rowResizeType: ResizeType.SELECTED,
      colResizeType: ResizeType.SELECTED,
    },
  },
  // totals: {
  //   col: {
  //     showGrandTotals: true,
  //     showSubTotals: false,
  //     reverseGrandTotalsLayout: true,
  //     reverseSubTotalsLayout: true,
  //     subTotalsDimensions: ['type'],
  //   },
  //   row: {
  //     showGrandTotals: true,
  //     showSubTotals: true,
  //     reverseGrandTotalsLayout: true,
  //     reverseSubTotalsLayout: true,
  //     subTotalsDimensions: ['province'],
  //   },
  // },
  // mergedCellsInfo: [
  //   [
  //     { colIndex: 1, rowIndex: 1, showText: true },
  //     { colIndex: 1, rowIndex: 2 },
  //   ],
  //   [
  //     { colIndex: 2, rowIndex: 1 },
  //     { colIndex: 2, rowIndex: 2, showText: true },
  //   ],
  // ],
  tooltip: S2TooltipOptions,
  style: {},
};

export const sliderOptions: SliderSingleProps = {
  min: 0,
  max: 10,
  step: 0.1,
  marks: {
    0.2: '0.2',
    1: '1 (默认)',
    2: '2',
    10: '10',
  },
};

export const s2ThemeConfig: ThemeCfg = {
  name: 'default',
  theme: {},
};

export const defaultOptions =
  getBaseSheetComponentOptions<SheetComponentOptions>(s2Options);
