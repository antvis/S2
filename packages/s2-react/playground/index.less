.playground {
  padding: 40px;

  .filter-section {
    margin: 20px 0;
    border: 1px solid #dcdcdc;
    padding: 20px;
  }

  .ant-collapse {
    margin-bottom: 20px;
  }

  .ant-checkbox-group {
    margin-left: 20px;
  }
}
