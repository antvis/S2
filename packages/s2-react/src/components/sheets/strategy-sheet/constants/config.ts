export enum KpiType {
  // 目标值
  TargetValue = 'TargetValue',
  // 基准值
  ReferenceValue = 'ReferenceValue',
  // 净增值
  NetAddedValue = 'NetAddedValue',
  // 净增目标值
  TargetNetAddedValue = 'TargetNetAddedValue',
  // 净增完成度
  NetAddedProgress = 'NetAddedProgress',
}

export const KPI_TYPES_CONFIG: Record<KpiType, { name: string; desc: string }> =
  {
    [KpiType.TargetValue]: {
      name: '目标值',
      desc: '指标目标字段的目标值；例如「指标名称1的目标值」',
    },
    [KpiType.ReferenceValue]: {
      name: '基准值',
      desc: '指标目标字段的基准值为（xxx），xxx为具体的数值；例如「指标名称1的基准值为（100000亿）」',
    },
    [KpiType.NetAddedValue]: {
      name: '净增值',
      desc: '指标目标字段的净增值，净增值=当前值-基准值；例如「指标名称1的净增值，净增为当前值-基准值」',
    },
    [KpiType.TargetNetAddedValue]: {
      name: '净增目标值',
      desc: '指标目标字段的净增目标，净增目标=目标值-基准值；例如「指标名称1的净增目标，净增为目标值-基准值」',
    },
    [KpiType.NetAddedProgress]: { name: '净增完成度', desc: '' },
  };
