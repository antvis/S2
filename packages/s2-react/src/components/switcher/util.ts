import { filter, flatten, isNil, map, mapValues } from 'lodash';
import { DraggableLocation } from 'react-beautiful-dnd';
import { getClassNameWithPrefix } from '@antv/s2';
import {
  FieldType,
  MAX_DIMENSION_COUNT,
  SWITCHER_FIELDS,
  SWITCHER_PREFIX_CLS,
} from './constant';
import {
  SwitcherItem,
  SwitcherResult,
  SwitcherState,
  SwitcherFields,
  SwitcherResultItem,
} from './interface';

export const getSwitcherClassName = (...classNames: string[]) =>
  getClassNameWithPrefix(SWITCHER_PREFIX_CLS, ...classNames);

export const getNonEmptyFieldCount = (fields: SwitcherFields) => {
  return SWITCHER_FIELDS.reduce(
    (sum, field) => sum + (isNil(fields[field]) ? 0 : 1),
    0,
  );
};

export const getMainLayoutClassName = (nonEmptyCount: number) => {
  switch (nonEmptyCount) {
    case 1:
      return getSwitcherClassName('content', 'one-dimension');
    case 2:
      return getSwitcherClassName('content', 'two-dimensions');
    default:
      return getSwitcherClassName('content', 'three-dimensions');
  }
};

export const shouldCrossRows = (nonEmptyCount: number, type: FieldType) =>
  nonEmptyCount < MAX_DIMENSION_COUNT || type === FieldType.Values;

export const moveItem = (
  source: SwitcherItem[],
  destination: SwitcherItem[],
  droppableSource: DraggableLocation,
  droppableDestination: DraggableLocation,
): SwitcherState => {
  // change order in same column
  if (droppableDestination.droppableId === droppableSource.droppableId) {
    const updatingDestination = [...destination];
    const [removed] = updatingDestination.splice(droppableSource.index, 1);
    updatingDestination.splice(droppableDestination.index, 0, removed);
    return {
      [droppableDestination.droppableId]: updatingDestination,
    };
  }
  // move to other column
  const updatingSource = [...source];
  const updatingDestination = [...destination];

  const [removed] = updatingSource.splice(droppableSource.index, 1);
  updatingDestination.splice(droppableDestination.index, 0, removed);

  return {
    [droppableSource.droppableId]: updatingSource,
    [droppableDestination.droppableId]: updatingDestination,
  };
};

export const checkItem = (
  source: SwitcherItem[],
  checked: boolean,
  id: string,
  parentId?: string,
): SwitcherItem[] => {
  const target: SwitcherItem = {
    ...source.find((item) => item.id === (parentId ?? id)),
  };

  // 有 parentId 时，说明是第二层级的改变
  if (parentId) {
    target.children = map(target.children, (item) => ({
      ...item,
      checked: item.id === id ? checked : item.checked,
    }));
  } else {
    target.checked = checked;
    target.children = map(target.children, (item) => ({
      ...item,
      checked,
    }));
  }

  return source.map((item) => (item.id === target.id ? target : item));
};

export const generateSwitchResult = (state: SwitcherState): SwitcherResult => {
  const generateFieldResult = (items: SwitcherItem[]): SwitcherResultItem => {
    const flattenValues = (list: SwitcherItem[]) =>
      flatten(
        map(list, ({ children, ...rest }) => {
          return [{ ...rest }, ...flattenValues(children)];
        }),
      );

    const allItems = flattenValues(items);

    //  get all hidden values
    const hideItems = filter(
      allItems,
      (item: SwitcherItem) => item.checked === false,
    );
    return {
      items: allItems,
      hideItems,
    };
  };

  return {
    [FieldType.Rows]: generateFieldResult(state[FieldType.Rows]),
    [FieldType.Cols]: generateFieldResult(state[FieldType.Cols]),
    [FieldType.Values]: generateFieldResult(state[FieldType.Values]),
  };
};

export const getSwitcherState = (fields: SwitcherFields): SwitcherState =>
  mapValues(fields, 'items');
