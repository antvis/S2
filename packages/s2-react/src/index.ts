// https://github.com/microsoft/TypeScript/issues/47663#issuecomment-1270716220
import type {} from 'rc-pagination';
import './utils/extendLocale';

export * from './components';
