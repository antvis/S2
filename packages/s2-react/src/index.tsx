// export * from '@antv/s2';
import React, { FC } from 'react';
import './index.less';
import './index.module.less';

export const Entry: FC = () => {
  return <div>Entry</div>;
};
