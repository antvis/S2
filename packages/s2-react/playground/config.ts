import { customMerge, isUpDataValue, S2DataConfig, S2Options } from '@antv/s2';
import type { SliderSingleProps } from 'antd';
import { isNil } from 'lodash';
import { getSheetComponentOptions } from '../src/utils';
import { data, totalData, meta } from '../__tests__/data/mock-dataset.json';

const BASIC_BACKGROUND_COLOR = '#FFFFFF';
const INTERACTIVE_BACKGROUND_COLOR = '#E1EAFE';

export const defaultTheme = {
  cornerCell: {
    icon: {
      size: 12,
    },
  },
  rowCell: {
    cell: {
      backgroundColor: BASIC_BACKGROUND_COLOR,
      interactionState: {
        hover: {
          backgroundColor: INTERACTIVE_BACKGROUND_COLOR,
        },
        selected: {
          backgroundColor: INTERACTIVE_BACKGROUND_COLOR,
        },
      },
    },
    icon: {
      size: 12,
    },
  },
  colCell: {
    cell: {
      interactionState: {
        hover: {
          backgroundColor: INTERACTIVE_BACKGROUND_COLOR,
        },
        selected: {
          backgroundColor: INTERACTIVE_BACKGROUND_COLOR,
        },
      },
      padding: {
        left: 4,
        right: 4,
      },
    },
  },
  dataCell: {
    cell: {
      crossBackgroundColor: BASIC_BACKGROUND_COLOR,
      interactionState: {
        hover: {
          backgroundColor: INTERACTIVE_BACKGROUND_COLOR,
        },
        hoverFocus: {
          backgroundColor: INTERACTIVE_BACKGROUND_COLOR,
        },
        selected: {
          backgroundColor: INTERACTIVE_BACKGROUND_COLOR,
        },
        unselected: {},
        prepareSelect: {
          borderColor: INTERACTIVE_BACKGROUND_COLOR,
        },
      },
    },
  },
};

export const tableSheetDataCfg: S2DataConfig = {
  data,
  totalData,
  meta,
  fields: {
    columns: ['province', 'city', 'type', 'sub_type', 'number'],
  },
};

export const pivotSheetDataCfg: S2DataConfig = {
  data,
  totalData,
  meta,
  fields: {
    rows: ['province', 'city'],
    columns: ['type', 'sub_type'],
    values: ['number'],
    valueInCols: true,
  },
};

export const s2Options: S2Options = {
  debug: true,
  width: 600,
  height: 400,
  hierarchyCollapse: false,
  interaction: {
    enableCopy: true,
  },
};

export const sliderOptions: SliderSingleProps = {
  min: 0,
  max: 10,
  step: 0.1,
  marks: {
    0.2: '0.2',
    1: '1 (默认)',
    2: '2',
    10: '10',
  },
};

export const strategyOptions: S2Options = {
  width: 1000,
  height: 400,
  cornerText: '指标',
  headerActionIcons: [
    {
      iconNames: ['Trend'],
      belongsCell: 'rowCell',
      defaultHide: true,
      action: () => {},
    },
  ],
  style: {
    cellCfg: {
      valuesCfg: {
        originalValueField: 'originalValues',
        conditions: {
          text: {
            field: 'number',
            mapping: (value, cellInfo) => {
              const { meta } = cellInfo;
              const isNormalValue = isNil(value);

              if (meta.fieldValue.values[0][0] === value || isNormalValue) {
                return {
                  fill: '#000',
                };
              }
              return {
                fill: isUpDataValue(value) ? '#FF4D4F' : '#29A294',
              };
            },
          },
        },
      },
    },
  },
};

export const mockGridAnalysisOptions: S2Options = {
  width: 1600,
  height: 600,
  style: {
    layoutWidthType: 'colAdaptive',
    cellCfg: {
      width: 400,
      height: 100,
      valuesCfg: {
        widthPercentCfg: [40, 20, 20, 20],
        conditions: {
          text: {
            field: 'number',
            mapping: (value, cellInfo) => {
              const { colIndex } = cellInfo;
              if (colIndex <= 1) {
                return {
                  fill: '#000',
                };
              }
              return {
                fill: isUpDataValue(value) ? '#FF4D4F' : '#29A294',
              };
            },
          },
        },
      },
    },
  },
};

export const defaultOptions: S2Options = customMerge(
  s2Options,
  getSheetComponentOptions({
    tooltip: {
      operation: {
        sort: true,
        tableSort: true,
        trend: true,
        hiddenColumns: true,
      },
    },
  }),
);
