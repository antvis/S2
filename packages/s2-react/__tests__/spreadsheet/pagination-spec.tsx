import {
  SpreadSheet,
  setLang,
  type LangType,
  type Pagination,
  type S2DataConfig,
} from '@antv/s2';
import { waitFor } from '@testing-library/react';
import React from 'react';
import type { Root } from 'react-dom/client';
import { pivotSheetDataCfg } from '../../playground/config';
import { SheetComponent, type SheetComponentsProps } from '../../src';
import * as mockDataConfig from '../data/simple-data.json';
import { renderComponent } from '../util/helpers';

const s2Options: SheetComponentsProps['options'] = {
  width: 600,
  height: 200,
  pagination: {
    current: 1,
    pageSize: 1,
  },
  hierarchyType: 'grid',
};

let s2: SpreadSheet;

describe('Pagination Tests', () => {
  let unmount: Root['unmount'];

  afterEach(() => {
    unmount?.();
  });

  // https://github.com/antvis/S2/issues/1697
  test.each([
    {
      locale: 'zh_CN',
      page: '1 条/页',
      count: '共计2条',
    },
    {
      locale: 'en_US',
      page: '1 / page',
      count: 'Total2',
    },
  ] as Array<{ locale: LangType; page: string; count: string }>)(
    'should render locale text for %o',
    async ({ locale, page, count }) => {
      setLang(locale);

      let spreadsheet: SpreadSheet;

      unmount = renderComponent(
        <SheetComponent
          options={s2Options}
          dataCfg={mockDataConfig as S2DataConfig}
          showPagination
          onMounted={(instance) => {
            spreadsheet = instance;
          }}
        />,
      );

      await waitFor(() => {
        expect(spreadsheet).toBeDefined();
        expect(
          document.querySelector('.ant-select-selection-item')?.innerHTML,
        ).toEqual(page);
        expect(
          document.querySelector('.antv-s2-pagination-count')?.innerHTML,
        ).toEqual(count);
      });
    },
  );

  test('should receive antd <Pagination/> component extra props', async () => {
    let spreadsheet: SpreadSheet;

    unmount = renderComponent(
      <SheetComponent
        options={{
          ...s2Options,
          pagination: {
            ...s2Options.pagination,
            current: 2,
            showSizeChanger: false,
            showQuickJumper: true,
          } as Pagination,
        }}
        dataCfg={mockDataConfig as S2DataConfig}
        showPagination
        onMounted={(instance) => {
          spreadsheet = instance;
        }}
      />,
    );

    await waitFor(() => {
      expect(spreadsheet).toBeDefined();
      expect(
        document.querySelector('.ant-pagination-options-quick-jumper'),
      ).toBeTruthy();
      expect(
        document.querySelector('.ant-pagination-options-size-changer'),
      ).toBeFalsy();
    });
  });

  test('should row header cell render text position based on the actual cell height when pagination is show', async () => {
    renderComponent(
      <SheetComponent
        options={{
          ...s2Options,
          pagination: {
            ...s2Options.pagination,
            current: 1,
            pageSize: 1,
          },
          height: 400,
        }}
        dataCfg={pivotSheetDataCfg}
        onMounted={(instance) => {
          s2 = instance;
        }}
        showPagination
      />,
    );

    await waitFor(() => {
      const rowCell = s2.facet.getRowCells()[0];

      expect(rowCell.getTextShape().parsedStyle.y).toBe(15);
    });
  });
});
