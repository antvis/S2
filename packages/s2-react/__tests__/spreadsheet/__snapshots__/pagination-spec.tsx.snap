// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Tests should render with antd <Pagination/> component 1`] = `
<DocumentFragment>
  <div
    class="antv-s2-wrapper"
  >
    <div
      class="antv-s2-container"
      style="position: relative;"
    >
      <canvas
        height="400"
        style="width: 600px; height: 200px; touch-action: none; display: block; user-select: none;"
        width="1200"
      />
    </div>
    <ul
      class="ant-pagination ant-pagination-mini css-dev-only-do-not-override-49qm"
    >
      <li
        class="ant-pagination-total-text"
      >
        共计 0 条
      </li>
      <li
        aria-disabled="true"
        class="ant-pagination-prev ant-pagination-disabled"
        title="Previous Page"
      >
        <button
          class="ant-pagination-item-link"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            aria-label="left"
            class="anticon anticon-left"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="left"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
              />
            </svg>
          </span>
        </button>
      </li>
      <li
        class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-disabled"
        tabindex="0"
        title="1"
      >
        <a
          rel="nofollow"
        >
          1
        </a>
      </li>
      <li
        aria-disabled="true"
        class="ant-pagination-next ant-pagination-disabled"
        title="Next Page"
      >
        <button
          class="ant-pagination-item-link"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            aria-label="right"
            class="anticon anticon-right"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
              />
            </svg>
          </span>
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`Pagination Tests should render with antd <Pagination/> component 2`] = `
<DocumentFragment>
  <div
    class="antv-s2-wrapper"
  >
    <div
      class="antv-s2-container"
      style="position: relative;"
    >
      <canvas
        height="400"
        style="width: 600px; height: 200px; touch-action: none; display: block; user-select: none;"
        width="1200"
      />
    </div>
    <ul
      class="ant-pagination ant-pagination-mini css-dev-only-do-not-override-49qm"
    >
      <li
        class="ant-pagination-total-text"
      >
        共计 2 条
      </li>
      <li
        aria-disabled="true"
        class="ant-pagination-prev ant-pagination-disabled"
        title="Previous Page"
      >
        <button
          class="ant-pagination-item-link"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            aria-label="left"
            class="anticon anticon-left"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="left"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
              />
            </svg>
          </span>
        </button>
      </li>
      <li
        class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
        tabindex="0"
        title="1"
      >
        <a
          rel="nofollow"
        >
          1
        </a>
      </li>
      <li
        class="ant-pagination-item ant-pagination-item-2"
        tabindex="0"
        title="2"
      >
        <a
          rel="nofollow"
        >
          2
        </a>
      </li>
      <li
        aria-disabled="false"
        class="ant-pagination-next"
        tabindex="0"
        title="Next Page"
      >
        <button
          class="ant-pagination-item-link"
          tabindex="-1"
          type="button"
        >
          <span
            aria-label="right"
            class="anticon anticon-right"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
              />
            </svg>
          </span>
        </button>
      </li>
      <li
        class="ant-pagination-options"
      >
        <div
          class="ant-pagination-options-quick-jumper"
        >
          Go to
          <input
            aria-label="Page"
            type="text"
            value=""
          />
          Page
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
